Bayes

DrPaulSharp opened the pull request:
I've made the following changes to the bayesResults struct so far:

1) Renamed bestPars to bestParams - I intend to make similar changes whenever they show up elsewhere.

2) Renamed bestFitsMean to bestFitMean and bestFitMean.ref to bestFitMean.reflectivity

3) Renamed predlims to predictionIntervals with fields reflectivity, sld, reflectivityXData, sldXData, sampleChi

4) Renamed parConfInts to confidenceIntervals with fields percentile95, percentile65, mean

5) Renamed nestOutput as nestedSamplerOutput

6) Removed Meas_info from dreamOutput

7) Moved bayesResults.dreamOutput.DREAMPar to bayesResults.dreamParams

8) Renamed dreamOutput.RunTime to dreamOutput.runtime

9) Renamed dreamParams.lambda to dreamParams.jumpProbability

I think we can go further, but let me know what you make of things so far: https://github.com/DrPaulSharp/RAT/tree/bayes
I would emphasise that we can easily make changes to what the user sees in their output without affecting the internals of the minimisers. I think one good example would be to change dreamParams.N to dreamParams.nChains, which matches the controls class.

In addition, I have a couple of questions:

1) In the example I have, results.bestFitPars is identical to bayesResults.bestParams. Is this always the case? If so we should remove bayesResults.bestParams from the output.

2) Is allChains only used by dream as seems to be the case? If so I suggest it is moved into dreamOutput.

StephenNneji commented:
@arwelHughes please see questions in PR

arwelHughes commented:
Hi Paul - yes to both questions! :) 
A

DrPaulSharp commented:
I'm currently looking at the fields of the `dreamParams` and `dreamOutput` structs. At the moment, they are the same as those set out in the paper: https://www.sciencedirect.com/science/article/abs/pii/S1364815215300396 However, I think we can use some better field names, by following the description column of the tables setting out the fields. For example:

![dream_params](https://github.com/RascalSoftware/RAT/assets/44529197/b63ff712-e5a5-4fa4-bb5c-ce7495ba0569)

Here, instead of `d`, `N`, and `T`, we could follow the description and use `nParams`, `nChains`, and `nGenerations` respectively. As well as being far more clear, this has the advantage that we can more closely match the dream parameters to those used in the controls class for the input (`nChains` is defined there for example) whilst retaining a clear link with how the paper sets out these fields.

What do we think? @arwelHughes @StephenNneji

arwelHughes commented:
That sounds reasonable to me, Paul...
A

________________________________
From: Paul Sharp ***@***.***>
Sent: Thursday, March 7, 2024 11:09 AM
To: RascalSoftware/RAT ***@***.***>
Cc: Hughes, Arwel (STFC,RAL,ISIS) ***@***.***>; Mention ***@***.***>
Subject: Re: [RascalSoftware/RAT] Refactors Bayes Results Struct (PR #212)


I'm currently looking at the fields of the dreamParams and dreamOutput structs. At the moment, they are the same as those set out in the paper: https://www.sciencedirect.com/science/article/abs/pii/S1364815215300396 However, I think we can use some better field names, by following the description column of the tables setting out the fields. For example:

dream_params.png (view on web)<https://github.com/RascalSoftware/RAT/assets/44529197/b63ff712-e5a5-4fa4-bb5c-ce7495ba0569>

Here, instead of d, N, and T, we could follow the description and use nParams, nChains, and nGenerations respectively. As well as being far more clear, this has the advantage that we can more closely match the dream parameters to those used in the controls class for the input (nChains is defined there for example) whilst retaining a clear link with how the paper sets out these fields.

What do we think? @arwelHughes<https://github.com/arwelHughes> @StephenNneji<https://github.com/StephenNneji>

—
Reply to this email directly, view it on GitHub<https://github.com/RascalSoftware/RAT/pull/212#issuecomment-1983281345>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGMNVXXKOFWW23QNCNXHWY3YXBDG5AVCNFSM6AAAAABEHNSC6CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBTGI4DCMZUGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

arwelHughes commented:
Oh, they won't have!
(but there are always a few! 🙂

________________________________
From: StephenNneji ***@***.***>
Sent: Thursday, March 7, 2024 11:19 AM
To: RascalSoftware/RAT ***@***.***>
Cc: Hughes, Arwel (STFC,RAL,ISIS) ***@***.***>; Mention ***@***.***>
Subject: Re: [RascalSoftware/RAT] Refactors Bayes Results Struct (PR #212)


I think its a good idea to change this except if >90% of our expected user have read the paper 😄 and might take offence

—
Reply to this email directly, view it on GitHub<https://github.com/RascalSoftware/RAT/pull/212#issuecomment-1983296432>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGMNVXSQESB6PPQFPE2VQFDYXBEKXAVCNFSM6AAAAABEHNSC6CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSOBTGI4TMNBTGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

DrPaulSharp commented:
Ok, I've made the following further changes:

1) Removed `bayesResults.bestParams` from the output

2) Moved `bayesResults.allChains` into `bayesResults.dreamOutput`

3) Renamed `dreamParams.d`  to `dreamParams.nParams`

4) Renamed `dreamParams.N` to `dreamParams.nChains`

5) Renamed `dreamParams.T` to `dreamParams.nGenerations` 

6) Renamed `dreamParams.epsilon` to `dreamParams.ABCCutoff` 

7) Renamed `dreamParams.modout` to `dreamParams.storeOutput`

I've also got some more questions:

1) How should we rename `dreamParams.delta`? The description is "Number chain pairs proposal" - would `nChainPairs` work?

2) What does `dreamParams.steps` do?

3) Both `dreamParams` and `dreamOutput` have outlier fields. Are the values of these related?

4) What does `dreamOutput.iloc` do?

5) How should we rename `dreamParams.R`? If I'm reading rightly, this affects the choices for the `dreamOutput` fields: `R_stat`, `AR`, and `CR`.

6) How should we rename `dreamOutput.fx`?

I've pushed the updates, all suggestions welcome. @arwelHughes @StephenNneji

StephenNneji commented on minimisers/DREAM/runDREAM.m:
```diff
@@ -47,42 +47,36 @@
 % This will change...
 % Initial sampling and parameter range
-Par_info.prior = coderEnums.priorTypes.Uniform;           
+Par_info.prior = coderEnums.priorTypes.Uniform;
```
should be parInfo or maybe paramInfo

StephenNneji commented on minimisers/generalUtils/bayesStats/findPosteriorsMax.m:
```diff
@@ -1,4 +1,4 @@
-function [bestPars_max,posteriors] = findPosteriorsMax(chain)
```
bestParams_max same as above

StephenNneji commented on minimisers/DREAM/runDREAM.m:
```diff
@@ -47,42 +47,36 @@
-lChains = DREAMPar.T;
-nPars = DREAMPar.d;
+nChains = DREAMPar.nChains;
+nPars = DREAMPar.nParams;
```
nParams instead of nPars

StephenNneji commented on minimisers/generalUtils/bayesStats/prctileConfInts.m:
maybe we should rename the function it looks like a typo?

DrPaulSharp replied:
Turns out "prctile" is a matlab function. I've renamed as `percentileConfidenceIntervals`

StephenNneji commented on minimisers/generalUtils/makeEmptyBayesResultsStruct.m:
```diff
@@ -2,29 +2,31 @@
+    %         bestFitMean: [1×1 struct]
+    % predictionIntervals: [1×1 struct]
+    % confidenceIntervals: [1×1 struct]
+    %           allChains: [1×1 struct]
```
any relation between chain and allChains?

DrPaulSharp replied:
Hmm, they're very large numerical arrays of different sizes and shapes. Might be worth asking about this one before doing anything.

StephenNneji commented on minimisers/generalUtils/makeEmptyBayesResultsStruct.m:
```diff
@@ -53,120 +55,119 @@
+                       'epsilon', 0.0250,...
+                              'IO', false,...
+                     'storeOutput', false,...
+                         'restart', false,...
```
I thought we were getting rid of restart and save?
